torchreid/models/__init__.py

```diff
@@ -74,13 +66,15 @@ def build_model(name, **kwargs):
         >>> model = models.build_model('resnet50', 751, loss='softmax')
     """
     avai_models = list(__model_factory.keys())
-    if name not in avai_models:
-        raise KeyError('Unknown model: {}. Must be one of {}'.format(name, avai_models))
     if name.startswith('q2l'):
         backbone_name = name[4:]
+        if backbone_name not in avai_models:
+            raise KeyError('Unknown backbone for Q2L model: {}. Must be one of {}'.format(backbone_name, avai_models))
         backbone = __model_factory[backbone_name](**kwargs)
         transformer = build_transformer(**kwargs)
         model = build_q2l(backbone, transformer, **kwargs)
```

Comment:
When q2l model is created, the original classifier is still exists in the backbone, so ptflops will return overestimated number of parameters. It's hard to fix, since all models are different, but we should keep that in mind.

Reply:
we could add an extra parameter, that would build model for feature extraction only, and for the classifier head, it would pass `nn.Identity`. But, we may keep it as is, of course since at the last meeting we have decided to not include this in SC SDC